reports-backend/src/test/java/org/jboss/da/reports/impl/ReportsGeneratorImplTest.java

```diff
@@ -246,6 +249,36 @@ public void testGetMultipleReport() throws CommunicationException, FindGAVDepend
         assertMultipleDependencies(report.getDependencies());
     }
 
+    /**
+     * Test the distinct on a stream in #getLookupReportsForGavs works correctly
+     */
+    @Test
+    public void testDistinctOnGavsStream() {
+        // Given
+        List<GAV> gavs = new ArrayList<>();
+        gavs.add(new GAV("org", "test", "1.0"));
+        gavs.add(new GAV("org", "test", "1.1"));
+        gavs.add(new GAV("org", "test", "1.0"));
+
+        gavs.add(new GAV("org2", "test2", "2.0"));
+        gavs.add(new GAV("org2", "test2", "2.2"));
+        gavs.add(new GAV("org2", "test2", "2.0"));
+        LookupGAVsRequest request = new LookupGAVsRequest(new HashSet<>(), new HashSet<>(), gavs);
+
+        // When
+        List<GAV> distinctedList = new ArrayList<>();
+        request.getGavs().stream()
+                .distinct()
+                .forEach((gav) -> {
+                    System.out.println("GAV: " + gav);
+                    distinctedList.add(gav);
+        });
+
+        //Then
+        assertEquals(new HashSet<>(gavs).size(), distinctedList.size());
+        assertTrue(new HashSet<>(gavs).containsAll(distinctedList));
```

Comment:
This doesn't test if distinct GAVs are filtered correctly, but whether Stream.distinct() and HashSet works the same.
If there is broken equals & hashCode on GAV, this can still pass.